applylib/applyset/results.go

```diff
@@ -49,6 +51,7 @@ func (r *ApplyResults) AllHealthy() bool {
 
 // checkInvariants is an internal function that warns if the object doesn't match the expected invariants.
 func (r *ApplyResults) checkInvariants() {
+	// TODO: Check the pruning results.
```

Comment:
TBH this is less important.  If you can think of a good sanity-check on the prune counters, great, but as this depends on the prior state I don't think there is one.